Backend/marketuserpost.js

```diff
@@ -89,6 +89,62 @@ async function retrieveMarketPosts(req, res) {
     }
 }
 
+/**
+ * Function to handle city filtering on users post
+ */
+async function filterMarketPosts(req, res) {
+    try {
+        const { city, state } = req.body;
+
+         // If neither city nor state is provided, return a specific message
+        if (!city && !state) {
+            return res.status(200).json({ message: 'No posts found. Please provide city or state parameters.' });
+        }
+
+        let query = {};
+    
+        // Check if the city parameter is provided
+        if (city) {
+          // Use case-insensitive regex for partial matching
+            query.city = new RegExp(city, 'i');
```

Comment:
Instead of using a regular expression for matching, consider using MongoDB's $regex operator, which is specifically designed for this purpose.